File: WolfDen/Lobbies/Slot.cs
using System;
using WolfDen.Lobbies.LobbyBehaviour;

namespace WolfDen.Lobbies
{
    public class Slot
    {
        public string Class { get { return GetType().Name; } }
        public Guid Id { get; } = Guid.NewGuid();
        public int Team { get; set; }
        public LobbyPlayer LobbyPlayer { get; set; }

        public Slot(int team) {
            Team = team;
        }

        
    }
}

File: WolfDen/wwwroot/ts/Lobby/Slot.ts
namespace TurnBasedBoardGame {
    export class Slot {
        Id: string;
        Team: number;
        SlotOptions: any;
        Lobby: Lobby;
        _lobbyPlayer: LobbyPlayer;

        set LobbyPlayer(value) {
            this._lobbyPlayer = value;
            if (value != null) {
                $("#slot" + this.Index + " .slotName h3").html(value.Name);
            }
            else {
                $("#slot" + this.Index + " .slotName h3").html("--Empty--");
            }
        }

        get LobbyPlayer(): LobbyPlayer{

            return this._lobbyPlayer;
        }

        get Index(): number {
            return Lobby.Instance.Slots.findIndex(slot => slot.Id == this.Id) ?? -1;
        }

        get IsCurrentPlayer(): boolean {
            return Lobby.Instance.GetCurrentSlot() === this;
        }

        constructor() {
        }

        Initialize() {
            var self = this;
            if (this.IsCurrentPlayer) {
                $("#positionNumber b").html(self.Team + "#");
            }
            $("#slot" + self.Index + " .slotTeam").html(self.Team + "#");
        }
    }
}